multidownloader/storage/storage_block.go

```diff
@@ -0,0 +1,142 @@
+package storage
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+
+	dbtypes "github.com/agglayer/aggkit/db/types"
+	aggkittypes "github.com/agglayer/aggkit/types"
+	"github.com/jmoiron/sqlx"
+	"github.com/russross/meddler"
+)
+
+type Blocks struct {
+	Headers  map[uint64]*aggkittypes.BlockHeader
+	AreFinal map[uint64]bool
+}
+
+func NewBlocks() Blocks {
+	return Blocks{
+		Headers:  make(map[uint64]*aggkittypes.BlockHeader),
+		AreFinal: make(map[uint64]bool),
+	}
+}
+
+func (b *Blocks) Add(header *aggkittypes.BlockHeader, isFinal bool) {
+	b.Headers[header.Number] = header
+	b.AreFinal[header.Number] = isFinal
+}
+
+func (b *Blocks) Get(number uint64) (*aggkittypes.BlockHeader, bool, error) {
+	header, exists := b.Headers[number]
+	if !exists {
+		return nil, false, fmt.Errorf("db.blocks.header: block header not found for number %d", number)
+	}
+	isFinal, exists := b.AreFinal[number]
+	if !exists {
+		return nil, false, fmt.Errorf("db.blocks.header: block finality not found for number %d", number)
+	}
+	return header, isFinal, nil
+}
+
+func (b *Blocks) ListHeaders() []*aggkittypes.BlockHeader {
+	headers := make([]*aggkittypes.BlockHeader, 0, len(b.Headers))
+	for _, header := range b.Headers {
+		headers = append(headers, header)
+	}
+	return headers
+}
+
+func (b *Blocks) IsEmpty() bool {
+	return len(b.Headers) == 0
+}
+
+func (a *MultidownloaderStorage) SaveBlockAggkitBlock(tx dbtypes.Querier,
```

Comment:
```suggestion
func (a *MultidownloaderStorage) SaveAggkitBlock(tx dbtypes.Querier,
```

Also, this is only used in tests, so I suggest making this a private function, or just update the `db` directly in the test.